src/pal/src/misc/corefx.cpp

```diff
@@ -346,3 +348,68 @@ ForkAndExecProcess(
     return success ? 0 : -1;
 }
 
+void CopyStat(struct corefx_stat* dst, const struct stat* src)
+{
+    dst->mode = src->st_mode;
+    dst->uid = src->st_uid;
+    dst->gid = src->st_gid;
+    dst->size = src->st_size;
+    dst->atime = src->st_atime;
+    dst->mtime = src->st_mtime;
+    dst->ctime = src->st_ctime;
+}
+
+int
+PALAPI
+GetStatisticsFromPath(
+    const char* path,
+    struct corefx_stat* buf)
+{
+    PERF_ENTRY(GetStatisticsFromPath);
+    ENTRY("GetStatisticsFromPath(path=%p (%s), buf=%p\n",
+          path, path ? path : "NULL", 
+          buf);
+
+    int success = FALSE;
+    struct stat result;
+
+    int ret = stat(path, &result);
+
+    if (ret == 0)
+    {
+        CopyStat(buf, &result);
+        success = TRUE;
+    } 
```

Comment:
We can, if you'd like. In practice, this is going to be coming from managed code and already zero'd.